tests/pyoozie/test_builder.py

```diff
@@ -131,8 +133,25 @@ def test_workflow_builder():
         kill_on_error='Failure message',
     )
 
+    # Is this XML expected
     actual_xml = builder.build()
-    assert xml_to_dict_unordered(expected) == xml_to_dict_unordered(actual_xml)
+    assert xml_to_dict_unordered(expected_xml) == xml_to_dict_unordered(actual_xml)
+
+    # Does it validate against the workflow XML schema?
+    try:
+        filename = tmpdir.join("workflow.xml")
+        filename.write_text(actual_xml, encoding='utf8')
+        subprocess.check_output(
+            'java -cp lib/oozie-client-4.1.0.jar:lib/commons-cli-1.2.jar '
+            'org.apache.oozie.cli.OozieCLI validate {path}'.format(path=str(filename)),
+            stderr=subprocess.STDOUT,
+            shell=True
+        )
+    except subprocess.CalledProcessError as e:
+        if request.config.getoption('verbose') > 0:
+            print(actual_xml)
+            print(e.output)
+        raise AssertionError('Invalid XML')
```

Comment:
I imagine we'll want to extra this into a helper function later and call it on each xml flow generated.